### PR TITLE
feat: auto push store paths and add CI

### PR DESCRIPTION
**Pull request by getchoo**
this does a few cool things

- auto push paths similar to https://github.com/cachix/cachix-action
- the option to not push any paths
- a test workflow (**REQUIRES SECRETS**)
- more informative logs
- better error catching
- the ability to use the attic cache without manually setting substituters
- prettier :nyaboom:
- a devshell

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          echo "STORE_PATH=$(pnpm store path)" >> "$GITHUB_OUTPUT"
+
+      - name: Setup pnpm cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
```
Comment (ryanccn) [inline .github/workflows/test.yaml:33]: `setup-node` has built-in support for pnpm caching

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - uses: cachix/install-nix-action@v22
+
+      - name: Setup Attic Cache
+        uses: ./
+        with:
+          endpoint: ${{ secrets.ATTIC_ENDPOINT }}
+          cache: ${{ secrets.ATTIC_CACHE }}
+          token: ${{ secrets.ATTIC_TOKEN }}
+
+      - run: nix-build test.nix
```
Comment (ryanccn) [inline .github/workflows/test.yaml:47]: nix CLI?

File: pnpm-lock.yaml
Comment (ryanccn) [inline pnpm-lock.yaml:1]: why format the lockfile, pnpm will switch it back to single quotes

File: src/stages/configure.ts [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+  try {
+    const endpoint = getInput("endpoint");
+    const cache = getInput("cache");
+    const token = getInput("token");
+
+    info("Logging in to attic cache");
+    await exec("attic", ["login", "--set-default", cache, endpoint, token]);
+    // remember existing source paths
+    info("Collecting store paths before build");
+    await exec(`${__dirname}/list-nix-store.sh > /tmp/store-path-pre-build`);
```
Comment (ryanccn) [inline src/stages/configure.ts:16]: Can this be implemented in the Node.js script directly?

**Review by ryanccn — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml (hunk 2 above)
Comment (getchoo) [reply .github/workflows/test.yaml]: `test.nix` is *very* impure, so it makes sense to use the old cli

File: pnpm-lock.yaml
Comment (getchoo) [reply pnpm-lock.yaml:1]: i didn't mean to, pnpm from the dev shell just did

File: src/stages/configure.ts (hunk 3 above)
Comment (getchoo) [reply src/stages/configure.ts:16]: yeah, i just thought this was a lot easier - plus it follows what cachix does which is good enough ™️

File: .github/workflows/test.yaml (hunk 2 above)
Comment (ryanccn) [reply .github/workflows/test.yaml]: I see

**Pull request merged**